From 7e5fa7c32683af137993e4161e89d0ed90ccf96c Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Wed, 23 Jul 2014 13:26:59 -0400
Subject: [input] synaptics: add manual min/max quirk for ThinkPad X240

Message-id: <1406122030-5632-4-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 86344
O-Subject: [RHEL7.1 / RHEL7.0.z PATCH 03/14] Input: synaptics - add manual min/max quirk for ThinkPad X240
Bugzilla: 1093449
RH-Acked-by: Aristeu S. Rozanski F. <arozansk@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Peter Hutterer <peter.hutterer@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1093449
Upstream: since v3.14, backported in stable
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7722437
Test: tested by myself on a t440s and a t410, was extensively tested in RHEL 6.6

commit 8a0435d958fb36d93b8df610124a0e91e5675c82
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Mar 28 01:01:38 2014 -0700

    Input: synaptics - add manual min/max quirk for ThinkPad X240

    This extends Benjamin Tissoires manual min/max quirk table with support for
    the ThinkPad X240.

    Cc: stable@vger.kernel.org
    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index ef148f9..d8d49d1 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -1507,6 +1507,14 @@ static const struct dmi_system_id min_max_dmi_table[] __initconst = {
   .driver_data = (int []){1024, 5052, 2258, 4832},
  },
  {
+  /* Lenovo ThinkPad X240 */
+  .matches = {
+   DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+   DMI_MATCH(DMI_PRODUCT_VERSION, "ThinkPad X240"),
+  },
+  .driver_data = (int []){1232, 5710, 1156, 4696},
+ },
+ {
   /* Lenovo ThinkPad T440s */
   .matches = {
    DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
-- 
1.7.1